keras/src/ops/core.py

```diff
@@ -1173,3 +1173,69 @@ def grad(*args, upstream):
     ```
     """
     return backend.core.custom_gradient(f)
+
+
+class Remat(Operation):
+    def __init__(self):
+        super().__init__()
+
+    def call(self, func, *args, **kwargs):
+        remat_func = backend.core.remat(func)
+        return remat_func(*args, **kwargs)
+
+    def compute_output_spec(self, f, *args, **kwargs):
+        return backend.compute_output_spec(f, *args, **kwargs)
```

Comment:
I think `func` should be a class member, not an argument to `call` (it's not a tensor).

```python
class Remat(Operation):
    def __init__(self, func):
        super().__init__()
        self.func = func

    def call(self, *args, **kwargs):
        remat_func = backend.core.remat(self.func)
        return remat_func(*args, **kwargs)

    def compute_output_spec(self, *args, **kwargs):
        return backend.compute_output_spec(self.func, *args, **kwargs)
```

Reply:
Not implementing this as a subclass of Operation. So removing this and updating 
def remat(f) instead of def remat(f, *args, **kwargs)
Thank you for the pointer!